router/downloader/downloader.go

```diff
@@ -163,6 +163,17 @@ func (dl *Download) Execute() error {
 		return ErrDownloadFailed
 	}
 	defer res.Body.Close()
+
+	//If the http code is a redirect get the location and set the url to that
```

Comment:
Since it uses recursion and the execute function first checks if the URL is in the local network it should throw an error if its in the local network.